v3/src/components/graph/adornments/lsrl/lsrl-adornment-model.ts

```diff
@@ -50,30 +61,29 @@ export const LSRLAdornmentModel = AdornmentModel
     const casesInPlot = dataConfig.subPlotCases(cellKey)
     const caseValues: Point[] = []
     casesInPlot.forEach((c: ICase) => {
-      const caseValueX = Number(dataset?.getValue(c.__id__, xAttrId))
-      const caseValueY = Number(dataset?.getValue(c.__id__, yAttrId))
+      const caseValueX = dataset?.getNumeric(c.__id__, xAttrId)
+      const caseValueY = dataset?.getNumeric(c.__id__, yAttrId)
       const caseValueLegend = dataset?.getValue(c.__id__, legendAttrId)
-      if (Number.isFinite(caseValueX)) {
-        if (cat && cat !== "__main__" && caseValueLegend !== cat) return
+      const isValidX = caseValueX && Number.isFinite(caseValueX)
+      const isValidY = caseValueY && Number.isFinite(caseValueY)
+      const categoryMatch = cat === "__main__" || caseValueLegend === cat
+      if (isValidX && isValidY && categoryMatch) {
         caseValues.push({x: caseValueX, y: caseValueY})
       }
     })
     return caseValues
   }
 }))
 .views(self => ({
-  confidenceValues(iX: number, caseValues: Point[], cellKey: Record<string, string>, lineIndex: number) {
+  confidenceValues(
+    iX: number, caseValues: Point[], cellKey: Record<string, string>, lineIndex: number, isInterceptLocked=false
+  ) {
     const lines = self.lines.get(self.instanceKey(cellKey))
     const line = lines?.[lineIndex]
-    const interceptIsLocked = false
-    const slopeIntercept = leastSquaresLinearRegression(caseValues, interceptIsLocked)
-    const { count, mse, xSumSquaredDeviations, xMean } = slopeIntercept
+    const { count, mse, xSumSquaredDeviations, xMean } = leastSquaresLinearRegression(caseValues, isInterceptLocked)
     if (
-      !line ||
-      (!count && count !== 0) ||
-      (!mse && mse !== 0) ||
-      (!xMean && xMean !== 0) ||
-      (!xSumSquaredDeviations && xSumSquaredDeviations !== 0)
+      !line || !line.intercept || !line.slope || count == null || mse == null || xMean == null ||
```

Comment:
`line.intercept` and `line.slope` can validly be 0, so the test should be `== null` for them as well.